### PR TITLE
Improved efficancy

### PR DESCRIPTION
**Pull request by Njpatman**
Made some tweaks, ready for database implementation.

### CODE REVIEW

File: addons/framework/statsTracking/fn_statsInitLocal.sqf [hunk 1]
```diff
@@ -11,15 +11,17 @@
...
 
 [0, 0, (getPlayerUID player), true] remoteExec ["CMF_fnc_updateStatArray", 2];
 
-player setVariable ["Shots_Fired", 0];
+player setVariable ["Player_UID", (getPlayerUID player)];
 
 player addEventHandler ["FiredMan", {
 	params ["_unit", "_weapon", "_muzzle", "_mode", "_ammo", "_magazine", "_projectile", "_vehicle"];
-	[_unit] call CMF_fnc_localShotsFired;
+	[] call CMF_fnc_localShotsFired;
```
Comment (tanaaKa) [inline addons/framework/statsTracking/fn_statsInitLocal.sqf]: `_shotsFired = _shotsFired + 1` here.

We cannot be doing calls every time someone fires.

File: addons/framework/statsTracking/fn_localShotsFired.sqf [hunk 2]
```diff
@@ -9,11 +9,4 @@
 
 */
```
Comment (tanaaKa) [inline addons/framework/statsTracking/fn_localShotsFired.sqf]: I would just get rid of this entire file. It's unnecessary.

File: addons/framework/statsTracking/fn_statsInitLocal.sqf [hunk 3]
```diff
@@ -56,4 +58,16 @@ switch true do
...
 		["inf_gi_roles", 0, (getPlayerUID player)] remoteExec ["CMF_fnc_updateStatArray", 2];
 	};
-};
\ No newline at end of file
+};
+
+[
+	{!alive player},
+	{
+		params ["_Player_UID"];
```
Comment (tanaaKa) [inline addons/framework/statsTracking/fn_statsInitLocal.sqf]: Use camelcase for variables and stop using underscores. An underscore should only begin a variable in sqf:

`_playerUID`

File: addons/framework/statsTracking/fn_statsInitLocal.sqf [hunk 4]
```diff
@@ -11,15 +11,17 @@
...
 waitUntil {not isNull player};
 
+Shots_Fired = 0;
+
 if (typeOf player isEqualTo "potato_spectate_playableSpectator" || typeOf player isEqualTo "VirtualCurator_F" || (getText (missionConfigFile >> "cmfVers") isEqualTo "") || (getText (getMissionConfig "Header" >> "gameType") isEqualTo "CMFSPCL")) exitWith {};
 
 [0, 0, (getPlayerUID player), true] remoteExec ["CMF_fnc_updateStatArray", 2];
 
-player setVariable ["Shots_Fired", 0];
+player setVariable ["Player_UID", (getPlayerUID player)];
```
Comment (tanaaKa) [inline addons/framework/statsTracking/fn_statsInitLocal.sqf]: Use camelcase for variables and stop using underscores. An underscore should only begin a variable in sqf:

`_playerUID`

File: addons/framework/statsTracking/fn_statsInitLocal.sqf [hunk 5]
```diff
@@ -11,15 +11,17 @@
 
 waitUntil {not isNull player};
 
+Shots_Fired = 0;
```
Comment (tanaaKa) [inline addons/framework/statsTracking/fn_statsInitLocal.sqf]: Use camelcase for variables and stop using underscores. An underscore should only begin a variable in sqf:

`_shotsFired`

File: addons/framework/statsTracking/fn_entityKilled.sqf [hunk 6]
```diff
@@ -18,27 +18,27 @@ switch true do
 {
 	//If a player kills a CIV
 	case (isPlayer _instigator && !(_sideKilled in ["potato_w", "potato_i", "potato_e"])):{
-		[1, 8, (getPlayerUID _instigator)] call CMF_fnc_updateStatArray;
+		[1, 8, (_instigator getVariable "Player_UID")] call CMF_fnc_updateStatArray;
```
Comment (tanaaKa) [inline addons/framework/statsTracking/fn_entityKilled.sqf]: Why this change? `getPlayerUID _instigator` is much faster than getting a variable.

Comment (Njpatman) [reply addons/framework/statsTracking/fn_entityKilled.sqf]: If a player dies, then their UID is no longer linked to that body. So we have to use a variable to identify the dead players UID.
